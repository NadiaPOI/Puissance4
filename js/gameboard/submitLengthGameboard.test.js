import { submitLengthGameboard } from './submitLengthGameboard.controller';

describe('submitLengthGameboard', () => {
  it('should generate a gameboard when submit form with rows and columns values', () => {
    document.body.innerHTML = `<button id="start">Start a Game</button>
    <aside>
      <form action="./index.html" method="post">
        <input type="text" name="row" id="row" value="6">
        <input type="text" name="col" id="col" value="7">
        <input type="submit" value="Generate gameboard">
      </form>
    </aside>`;

    submitLengthGameboard();

    let btn = document.querySelector('input[type="submit"]');
    btn.click();

    expect(document.getElementById('gameboard')).toMatchSnapshot();
  });
});
