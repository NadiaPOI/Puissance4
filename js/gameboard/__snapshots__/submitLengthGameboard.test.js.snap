// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitLengthGameboard should generate a gameboard when submit form with rows and columns values 1`] = `
<table
  id="gameboard"
>
  <tr
    id="0"
  >
    <td>
      <img
        class="0"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="1"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="2"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="3"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="4"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="5"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="6"
        src="./img/empty.png"
      />
    </td>
  </tr>
  <tr
    id="1"
  >
    <td>
      <img
        class="0"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="1"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="2"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="3"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="4"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="5"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="6"
        src="./img/empty.png"
      />
    </td>
  </tr>
  <tr
    id="2"
  >
    <td>
      <img
        class="0"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="1"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="2"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="3"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="4"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="5"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="6"
        src="./img/empty.png"
      />
    </td>
  </tr>
  <tr
    id="3"
  >
    <td>
      <img
        class="0"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="1"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="2"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="3"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="4"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="5"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="6"
        src="./img/empty.png"
      />
    </td>
  </tr>
  <tr
    id="4"
  >
    <td>
      <img
        class="0"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="1"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="2"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="3"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="4"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="5"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="6"
        src="./img/empty.png"
      />
    </td>
  </tr>
  <tr
    id="5"
  >
    <td>
      <img
        class="0"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="1"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="2"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="3"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="4"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="5"
        src="./img/empty.png"
      />
    </td>
    <td>
      <img
        class="6"
        src="./img/empty.png"
      />
    </td>
  </tr>
</table>
`;
